### ISSUE TITLE
Better feedback during the import process

### ISSUE THREAD

**Issue by jidanni** [umap-project/umap#564]
When Importing a .geojson file with 3000 points:
1. No spinner so one can tell when the upload is finished.
1. No documentation on how many points is the max allowed.
2. Maybe not all my points got uploaded... The browser became unresponsive.

**Comment by jidanni** [comment]
Now putting the file on a website, and entering the URL so as to upload from URL... same thing happens: nothing.

**Comment by Aurelie-Jallut** [comment]
I'm opening up the subject of user feedback.
Here are 2 observations linked to 2 user tests
- on the OSM instance: After importing the datamine water point data for a commune: it's slow and buggy
=> no error message

- remote import of geodatamine library data for Lyon / cadastre import (in layer and remotely) => ‘Problem in server response’ error message following import 

=> It's better to have a message even if you don't really understand whether it's temporary, whether you need to relay the problem somewhere or whether it will work later.

**Comment by yohanboniface** [comment]
If I summarize, we need:

- to start the loader when starting the export (and make the loader more visible ? => #148), and stop it at the end
- zoom to the imported data after import (imported data != datalayer data, as it may already have some preexisting data)
- to have a message when the import does not found any data

@Aurelie-Jallut thoughts ? :)

**Comment by yohanboniface** [comment]
A bit of technical considerations:
- what we need here is to be able to plug in the end of a whole data import, after which we could decide:
  - whether to create a datalayer or not
  - where to zoom
  - alert if not data has been found

The fact is that all the import process for now is blind about the imported content, and only the already created datalayer knows about the content, and when it knows it it is already mixed with preexisting content if any. Also  we need to keep in mind that it's possible to import multiple files at once (so it's like multiple import one after the other).

We think that a clean approach would be to totally extract the data management from the datalayer to a dedicated "importer" or whatever class. But this change is quite committing, and given we are about to integrate vector tiles and thus more concepts around layers, it may not be the best moment to refactor this part (yet). Let's keep this in mind when we work on integrating vector tiles.

In the mean time, we could have a temporary solution with either a promise returned by the import chain process (a bit intrusive), an event (but now uMap classes do not know about events, only Leaflet ones do), or maybe a very targeted layer function that would be called after each import whatever the caller of this import (eg. event when using a query string url).

**Comment by Aurelie-Jallut** [comment]
another question : )
What types of error can be detected when importing data?
(the idea would be to see if we could make the messages clearer depending on the problem encountered or see how to avoid them beforehand).

**Comment by Aurelie-Jallut** [comment]
To improve visibility while leaving the loader where it is, we could: 
- reduce the width of the coloured bar
- Increase the contrast

![Image](https://github.com/user-attachments/assets/1d4c6ae1-2772-4ecd-ba5e-cd2460fbc436)

**Comment by yohanboniface** [comment]
> @yohanboniface Can you give me an idea of the different times when the loader is displayed?

At this time, I think it is only displayed when there is an http request going on, whether a tile image we are waiting to or a remote data source fetch or an import from a remote URL. But we could start it at the beginning of the import as well, and stop it at the end.

>  What types of error can be detected when importing data?

Humm, it's quite complex.
Import can come from one file, multiple files (so with different formats and/or success/error status), one copy-paste, one URL. For each we can have various subissues, but at this time we are not catching everything in details. Plus each of those source can lead to errors at various moment: the copy-paste is immediately processed, but an URL first need to be fetched, and a file to be loaded.
We worked on this topic yesterday, and we tried to at least get an error message when error, an other error message when empty, and a success message (with a counter) when all good. Should we go in greater details ?

> To improve visibility while leaving the loader where it is, we could:
>    reduce the width of the coloured bar
>    Increase the contrast

Can you be a bit more specific? I'm not sure what I should do to implement such improvements :)
Like: which width to reduce ? what colour to use where to increase the contrast ? :)

**Comment by Aurelie-Jallut** [comment]
> We worked on this topic yesterday, and we tried to at least get an error message when error, an other error message when empty, and a success message (with a counter) when all good. Should we go in greater details ?

Yes, I saw that! Well done ! It's a great improvement! We could work on messages. For example, if we know that the problem is that the file is too large, we can put a message giving this information, if we know that the server crashed because of our fault, we could also tell it...

> Can you be a bit more specific? I'm not sure what I should do to implement such improvements :) Like: which width to reduce ? what colour to use where to increase the contrast ? :)

- [width] Reduce the green band to make it more visible. I'd test with a width of 1/4 of the screen.
- [contrast] You could darken the background of the loader (I put an image below to clarify)
- [color] Use the accent colour: 42ECE6

![Image](https://github.com/user-attachments/assets/c6783201-ae40-440f-b85d-6dad3b50847c)
